fs2/src/main/scala/io/finch/fs2/package.scala

```diff
@@ -68,14 +70,15 @@ trait StreamInstances {
         .compile
         .drain
 
-      F.toIO(run).unsafeRunAsyncAndForget()
-      p
+      F.bracketCase(F.start(run))(_ => F.pure(p: Reader[Buf])) {
```

Comment:
@sergeykolbasov @joroKr21 Please take a look.

Reply:
Sorry, I was really busy these days. This change LGTM but it's unfortunate that we have to ask for ConcurrentEffect since this is the most powerful typeclass in the hierarchy and permits the least amount of implementations (e.g. Scalaz Task can implement Effect but not ConcurrentEffect). I'm trying but failing to understand why evaluation of the Stream has to happen  here.

Reply:
Thanks a lot for looking into this. What we simply need to do is to run `run: F[Unit]` and forget about it (in a sense only worrying about cancelations) within [`ToService`](https://github.com/finagle/finch/blob/master/core/src/main/scala/io/finch/ToService.scala#L126-L143). So effectively we need to include `run: F[Unit]` evaluation as part of the effect we return here, from `ToResponse`, but not wait for its result.

One way to do that is to `runAsyncAndForget` but we figured this rules out cancelation signals. Another way is to use `ConcurrentEffect` and fibers as we're basically racing two effects. I'm not sure there are other ways to express that but I'm wiling to spend more time researching (there is no rush on merging this).

Also, good point about the callback.